scenario/src/main/kotlin/com/cosmotech/scenario/azure/ScenarioServiceImpl.kt

```diff
@@ -896,7 +898,8 @@ internal class ScenarioServiceImpl(
   ): ScenarioAccessControl {
     val scenario = findScenarioByIdNoState(organizationId, workspaceId, scenarioId)
     csmRbac.verify(scenario.getRbac(), PERMISSION_WRITE_SECURITY, scenarioPermissions)
-    csmRbac.getAccessControl(scenario.getRbac(), identityId)
+    csmRbac.getAccessControlOrThrow(
+        scenario.getRbac(), identityId, "User '$identityId' not found in scenario $workspaceId")
```

Comment:
Following the dev review on cosmotech-api-common, use the correct method `(verify|check)Rights`